test/libsolidity/SolidityNameAndTypeResolution.cpp

```diff
@@ -208,7 +208,7 @@ BOOST_AUTO_TEST_CASE(external_structs)
 			struct X { bytes32 x; Test t; Simple[] s; }
 			function f(ActionChoices, uint, Simple) external {}
 			function g(Test, Nested) external {}
-			function h(function(Nested) external returns (uint)[]) external {}
```

Comment:
Shouldn't this be `memory`?

Reply:
I'm actually not sure.

Reply:
I'm actually wondering right now, why this needs to be changed for this PR at all... Is a nested function argument somehow considered a local variable declaration? That would seem rather weird to me... I should probably look into that and at least understand why the code change above even affects this.

Reply:
The scope of the variable declaration ``Nested`` seems to be the callable declaration of function ``h``. And ``VariableDeclaration::isLocalVariable()`` considers everything a local variable, whose scope is a callable declaration, a block or a for statement.
Do you think this makes sense? To me it seems that this case should be excluded there and that ``Nested`` here is **not** a local variable. If you agree I'll create an issue for it.

Reply:
For this PR I now changed it to ``memory`` in any case, although (as it is considered a local variable, but **not** a callable parameter in ``VariableDeclaration::isCallableParameter``) it defaulted to ``storage`` before.